fix(DB/Quest): Rework support for 'Someone Else's Hard Work Pays Off'

Nyeriah opened the pull request:
<!-- First of all, THANK YOU for your contribution. -->

## Changes Proposed:
-  
-  

## Issues Addressed:
<!-- If your fix has a relating issue, link it below -->
- Closes https://github.com/chromiecraft/chromiecraft/issues/4566

## SOURCE:
<!-- If you can, include a source that can strengthen your claim -->

## Tests Performed:
<!-- Does it build without errors? Did you test in-game? What did you test? On which OS did you test? Describe any other tests performed -->
- 
- 


## How to Test the Changes:
<!-- Describe in a detailed step-by-step order how to test the changes -->

1.
2.
3.

## Known Issues and TODO List:
<!-- Is there anything else left to do after this PR? -->

- [ ]
- [ ]

<!-- If you intend to contribute repeatedly to our project, it is a good idea to join our discord channel. We set ranks for our contributors and give them access to special resources or knowledge: https://discord.com/invite/DasJqPba)
     Do not remove the instructions below about testing, they will help users to test your PR -->
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

Nyeriah commented:
I've identified an issue with the second last power core thing, where the scort npc mentions some mobs/attack wave but nothing actually spawns. Otherwise this works, needs sniff to wrap up that last pack though.

Buggyswires commented:
Bug confirmed. I did the escort with a group and was the only one actually having the quest.
I didn't die, but the npc still just disappeared at the end without giving credit.

Nyeriah commented:
> Bug confirmed. I did the escort with a group and was the only one actually having the quest.
> I didn't die, but the npc still just disappeared at the end without giving credit.

This doesn’t relate to this PR I presume

Nyeriah commented:
Please test this and check if it works, it's still better than a fully broken quest I assume

Sneften commented:
Tested ✅ 
Got quest complete on 2 characters.

Only thing that's not working is when the groups spawn to ambush you, they just spawn and stand still. Won't enter combat unless you aggro them.
Blizz-like they spawn and run to the escort NPC "forcing" combat.

Gultask commented:
sql file can't be ran twice:
![image](https://user-images.githubusercontent.com/100873791/210436692-ed3c095f-0eba-47ac-8cbc-31d71f1fa5f2.png)

Nyeriah commented:
> sql file can't be ran twice: ![image](https://user-images.githubusercontent.com/100873791/210436692-ed3c095f-0eba-47ac-8cbc-31d71f1fa5f2.png)

Correct. SQL that alters table structure can't be ran twice and there's nothing we can do about it

Gultask commented:
That's unfortunate :c

Gultask reviewed (changes requested):
wait, sorry. It's missing deletes

Gultask commented on data/sql/updates/pending_db_world/rev_1671290232250846500.sql:
```diff
@@ -0,0 +1,336 @@
+(19666,1,0,"Your plan was a good one, Sha'heen, and you would have gotten away with it if not for one thing...",12,0,100,1,0,0,17243,0,"Shadow Lord Xiraxis"),
+(19666,2,0,"Never underestimate the other ethereal's greed!",12,0,100,15,0,0,17244,0,"Shadow Lord Xiraxis");
+
+DELETE FROM `waypoints` WHERE `entry` IN (1967100,1967101,1967102);
+INSERT INTO `waypoints` (`entry`,`pointid`,`position_x`,`position_y`,`position_z`,`point_comment`) VALUES
```
```suggestion
DELETE FROM `waypoints` WHERE `entry` IN (19666,1967100,1967101,1967102);
INSERT INTO `waypoints` (`entry`,`pointid`,`position_x`,`position_y`,`position_z`,`point_comment`) VALUES
```

acidmanifesto commented:
Suggestion. Mark as merge later. Then merge before squash day

Gultask commented:
> Tested ✅ Got quest complete on 2 characters.
> 
> Only thing that's not working is when the groups spawn to ambush you, they just spawn and stand still. Won't enter combat unless you aggro them. Blizz-like they spawn and run to the escort NPC "forcing" combat.

There's a flag set to make the group aggro the player immediately so I don't know why it's not firing. Maybe changing to script owner will fix it but I've had several problems with that as well

You can try that by changing like 70 to:
```sql
(1967105,9,4,0,0,0,100,0,0,0,0,0,0,107,0,0,1,0,0,0,1,0,0,0,0,0,0,0,0,"Cryo-Engineer Sha'heen - On Script - Summon Group 0"),
```

Nyeriah commented:
The amount of time I've available to invest into this rages widely, plus it's SAI and I detest SAI. If anyone wants to try addressing the issues reported, feel free to, you can edit the file as you will. Otherwise I will return to it later, much later

Gultask commented:
Yeah, don't worry about it. I'm planning on testing it later too, seeing if changing the parameter works :)

Gultask commented on data/sql/updates/pending_db_world/rev_1671290232250846500.sql:
```diff
@@ -0,0 +1,336 @@
+(1967107,9,1,0,0,0,100,0,0,0,0,0,0,66,0,0,0,0,0,0,8,0,0,0,0,0,0,0,0.13962,"Cryo-Engineer Sha'heen - On Script - Set Orientation"),
+(1967107,9,2,0,0,0,100,0,1000,1000,0,0,0,1,10,0,0,0,0,0,12,1,0,0,0,0,0,0,0,"Cryo-Engineer Sha'heen - On Script - Say Line 10"),
+(1967107,9,3,0,0,0,100,0,0,0,0,0,0,17,233,0,0,0,0,0,1,0,0,0,0,0,0,0,0,"Cryo-Engineer Sha'heen - On Script - Set Emote State 233"),
+(1967107,9,4,0,0,0,100,0,1000,1000,0,0,0,107,1,1,0,0,0,0,1,0,0,0,0,0,0,0,0,"Cryo-Engineer Sha'heen - On Script - Summon Group 1"),
```
```suggestion
(1967107,9,4,0,0,0,100,0,1000,1000,0,0,0,107,1,0,1,0,0,0,1,0,0,0,0,0,0,0,0,"Cryo-Engineer Sha'heen - On Script - Summon Group 1"),
```

Gultask reviewed (commented):
This works

Nyeriah commented:
Squash days may or may not happen before the end of times, should we merge?

Gultask commented:
yeah
